src/components/HdBadge.vue

```diff
@@ -0,0 +1,217 @@
+<template>
+  <!-- collapsible badge component could have been structured in a semantic way using
+  html5 <details> & <summary> tags but unfortunately IE doesn't support them -->
+  <div
+    :class="computedClasses"
+  >
+    <span
+      class="badge__label"
+      @click.capture="collapse"
+    >
+      <template v-if="showIconBefore">
+        <img
+          v-if="icon"
+          :src="icon"
+          class="badge__icon"
+        >
+        <svg
+          v-else
+          class="badge__icon"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <circle
+            cx="7"
+            cy="7"
+            r="7"
+            fill="currentColor"
+          />
+        </svg>
+      </template>
+      {{ label }}
+    </span>
+    <p
+      v-if="hasDetails"
+      class="badge__details"
+    >
+      <slot />
+    </p>
+  </div>
+</template>
+
+<script>
+export const TYPES = {
+  DEFAULT: '',
+  PRIMARY: 'primary',
+  SECONDARY: 'secondary',
+  TERTIARY: 'tertiary',
+  DISABLED: 'disabled',
+};
+
+export default {
+  name: 'HdBadge',
+  props: {
+    modifier: {
+      type: String,
+      default: TYPES.DEFAULT,
+      validator(value) {
+        const allTypes = Object.values(TYPES);
+
+        return allTypes.includes(value);
+      },
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    showIconBefore: {
+      type: Boolean,
+      default: true,
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+  },
+  data() {
+    return {
+      collapsed: false,
+    };
+  },
+  computed: {
+    computedClasses() {
+      const baseClass = 'badge';
+      const classes = [baseClass];
+
+      if (this.modifier) {
+        classes.push(`${baseClass}--${this.modifier}`);
+      }
+      if (this.hasDetails) {
+        classes.push(`${baseClass}--with-details`);
+      }
+      if (this.collapsed) {
+        classes.push(`${baseClass}--collapsed`);
+      }
+
+      return classes;
+    },
+    hasDetails() {
+      return !!this.$slots.default;
+    },
+  },
+  methods: {
+    collapse() {
+      if (this.hasDetails) this.collapsed = !this.collapsed;
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.badge{
+    display: flex;
+    justify-content: flex-start;
+
+    &__label{
+      display: flex;
+      align-items: center;
+      font-weight: bold;
+      font-size: 18px;
+      line-height: 28px;
+      border-radius: 4px;
+      padding: $stack-xs $inline-s;
+      cursor: default;
+
+      &::after{
+        content: ' ';
+        display: none;
+        background: url('~hd-blocks/assets/icons/ic_chevron-down--small.svg');
```

Comment:
We can use the chevron we already have here (it won't make a big difference UI-wise):
```
background: url('~hd-blocks/assets/icons/ic_chevron-down.svg') no-repeat;
width: 16px;
height: 16px;
```